src/script/sign.cpp

```diff
@@ -26,6 +26,9 @@ bool MutableTransactionSignatureCreator::CreateSig(const SigningProvider& provid
     if (sigversion == SigVersion::WITNESS_V0 && !key.IsCompressed())
         return false;
 
+    // Signing for witness scripts needs the amount.
+    if (sigversion == SigVersion::WITNESS_V0 && amount < 0) return false;
```

Comment:
nit: might have been interesting to make it available here and use `return HandleMissingData(MissingDataBehavior::FAIL)` to make this easier to grep. But I can see that it's probably not worth the additional effort.